server/src/main/java/org/candlepin/resource/SubscriptionResource.java

```diff
@@ -219,19 +220,15 @@ public Response activateSubscription(
     @Path("/{subscription_id}")
     @Produces(MediaType.APPLICATION_JSON)
     public void deleteSubscription(@PathParam("subscription_id") String subscriptionId) {
+        // Look up master pool by subscription Id
+        Pool master = this.poolManager.getMasterPoolBySubscriptionId(subscriptionId);
 
-        // Lookup pools from subscription ID
-        int count = 0;
-
-        for (Pool pool : this.poolManager.getPoolsBySubscriptionId(subscriptionId)) {
-            this.poolManager.deletePool(pool);
-            ++count;
-        }
-
-        if (count == 0) {
+        if (master == null) {
             throw new NotFoundException(
                 i18n.tr("A subscription with the ID \"{0}\" could not be found.", subscriptionId));
         }
+
+        this.poolManager.deletePools(Collections.singletonList(master), null);
```

Comment:
This is pretty minor, but would you mind switching this to 

```
this.poolManager.deletePools(Collections.singletonList(master))
```

so we don't have that superfluous `null` hanging around.  I think the functionality should be the same since the one parameter method just delegates to the two parameter method but provides the `null` as the second parameter.

Reply:
But... extra method call overhead.

Changed.